### PR TITLE
LTS-2025: WEBUI-1934: handle single-select hierarchical vocabulary in saved search params

### PR DESCRIPTION
**Pull request by swarnadipa-dev**
https://hyland.atlassian.net/browse/WEBUI-1934

### CODE REVIEW

File: elements/search/nuxeo-search-form.js [hunk 1]
```diff
@@ -581,6 +581,15 @@ Polymer({
                     }
                     return output;
                   });
+                } else if (modifyPayload && value && typeof value === 'object' && value.id && value.properties) {
+                  // Single-select hierarchical vocabulary: reconstruct full path (e.g. "bankN1/normalCard")
+                  let output = value.id;
+                  let entry = value;
+                  while (entry && entry.properties && entry.properties.parent) {
+                    output = `${entry.properties.parent.id}`.concat('/', `${output}`);
+                    entry = entry.properties.parent;
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply elements/search/nuxeo-search-form.js]: Good catch — addressed in the latest commit (b761db3 / 184aad4).

Both the single-select branch and the existing array branch now extract `parentId` defensively:

```js
const parent = entry.properties.parent;
const parentId = typeof parent === 'string' ? parent : parent.id ?? parent?.properties?.id;
if (!parentId) break;
```

This handles:
- `properties.parent` as a plain string id
- `properties.parent` as an object with `.id`
- `properties.parent` as an object with `.properties.id`
- Stops traversal cleanly (no `undefined/...` paths) when no usable id is found

4 dedicated unit tests added in `test/nuxeo-search-form.test.js` to cover all these edge cases.

File: elements/search/nuxeo-search-form.js [hunk 2]
```diff
@@ -576,11 +576,26 @@ Polymer({
                   result[param] = value.map((item) => {
                     let output = item.id ? item.id : item;
                     while (item && item.properties && item.properties.parent) {
-                      output = `${item.properties.parent.id}`.concat('/', `${output}`);
-                      item = item.properties.parent;
+                      const parent = item.properties.parent;
+                      const parentId = typeof parent === 'string' ? parent : (parent.id ?? parent?.properties?.id);
+                      if (!parentId) break;
+                      output = `${parentId}`.concat('/', `${output}`);
+                      item = typeof parent === 'string' ? null : parent;
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply elements/search/nuxeo-search-form.js]: Fixed — added test cases for both defensive parent shapes:

- **(a)** `properties.parent` as a plain string id → `'parentStringId/child'`
- **(b)** `properties.parent` as an object whose id is under `parent.properties.id` → `'nestedParentId/child'`

These are covered for both single-select and array (multi-select) cases in the latest commit.

File: elements/search/nuxeo-search-form.js [hunk 3]
```diff
@@ -576,11 +576,26 @@ Polymer({
...
+                  let output = value.id;
+                  let entry = value;
+                  while (entry && entry.properties && entry.properties.parent) {
+                    const parent = entry.properties.parent;
+                    const parentId = typeof parent === 'string' ? parent : (parent.id ?? parent?.properties?.id);
+                    if (!parentId) break;
+                    output = `${parentId}`.concat('/', `${output}`);
+                    entry = typeof parent === 'string' ? null : parent;
+                  }
+                  result[param] = output;
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply elements/search/nuxeo-search-form.js]: Done — extracted `toHierarchicalPath(entry)` as a shared helper used by both the single-select object branch and the array (multi-select) branch. This ensures both code paths handle all parent shapes consistently and future fixes only need to be made in one place.

File: test/nuxeo-search-form.test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    const result = mutate(savedSearchParams, true);
+
+    expect(result).to.not.have.property('dc:title');
+    expect(result['my:vocabField']).to.equal('parentCategory/childItem');
+    expect(result.ecm_fulltext).to.equal('');
+    expect(result['cvd:contentViewName']).to.equal('my_search');
+  });
+
+  test('multi-select hierarchical vocab: each item reconstructed to path string; modifyPayload=false is a no-op', () => {
+    // Simulates saved search params for a multi-select hierarchical vocab field.
```
[the remark this answers is not in the document]
Comment (swarnadipa-dev) [reply test/nuxeo-search-form.test.js]: Fixed — the test name now reads: `'vocab objects without modifyPayload are not converted to path strings'`, making it explicit that the guard is specifically about hierarchical vocab objects/arrays not being transformed when `modifyPayload` is `false`.

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Review by AnilKumarVanga — APPROVED** [review]
